release-notes/major12/12.0.0/spring-boot3-migration.md

```diff
@@ -0,0 +1,70 @@
+- [Migrating to Spring Security 6](https://docs.spring.io/spring-security/reference/6.0/migration/index.html)
+
+As a result of the Spring Boot 3 upgrade, some other frameworks / dependencies needed to be upgraded as well. 
+Each chapter below describes the changes for the relevant framework or dependency.  
+
+## Spring Boot 3 / Spring 6
+
+- `spring.factories` have been [replaced](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#auto-configuration-files) by `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
+  - Listed configuration classes need to be annotated with `@AutoConfiguration`. More info can be found [here](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.7-Release-Notes#changes-to-auto-configuration)
+- `@Controller` or `@RestController` annotations are [now mandatory](https://github.com/spring-projects/spring-framework/issues/22154)
+  - When declaring the controller beans via autoconfiguration, please also annotate the controller with `com.ritense.valtimo.contract.annotation.@SkipComponentScan` to prevent duplicated beans.
+- `javax` packages [changed](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#jakarta-ee) to `jakarta`
+- `@ConstructingBinding` on  no longer supported/needed for `@ConfigurationProperties` classes.
+- Java 8 is no longer supported. Please use Java 17.
+
+## Spring Security 6
+
+- `HttpSecurity.authorizeRequests` has been deprecated. 
+  - When implementing the `HttpSecurityConfigurer`, this cannot be mixed with the replacement (`authorizeHttpRequests`).
+
+Before:
+```java
+http.authorizeRequests()
+    .antMatchers(GET, "/api/v1/ping").permitAll();
+```
+New:
+```java
+http.authorizeHttpRequests((requests) ->
+    requests.requestMatchers(antMatcher(GET, "/api/v1/ping")).permitAll()
+);
+```
+- `WebSecurityConfigurerAdapter` has been removed.
+  - Removed `CoreHttpSecurityConfigurerAdapter`. This has been replaced by `ValtimoCoreSecurityFactory`
+- `AuthorizedUrl.acccess(..)` no longer supports expressions as an argument
+  - `WhitelistIpRequest` has been replaced by `WhitelistIpRequestMatcher`. An example can be found in `CamundaCockpitHttpSecurityConfigurer`
+      
+- Cockpit is now whitelist-secured again. Support for CIDRs added to support a range of addresses.
+  - A list of hosts can be set at `valtimo.security.whitelist.hosts`
+- Removed `AuthenticationSecurityConfigurer`. No replacement other than implementing a `SecurityFilterChain` directly.
+- OpenAPI endpoint `/v3/api-docs` is now only available to `ROLE_DEVELOPER`
+
+## Hibernate 6
+
+- Overriding entity properties is no longer supported. As a result:
+  - `ProcessLink` implementations can no longer be a `data` class and cannot override the base properties of `ProcessLink`. 
+  Methods previously generated by the `data` type like `equals(other: Any?)`, `hashCode()` and `copy(...)` should be migrated accordingly.
```

Comment:
Shouldn't this have a dash as well? If not, then it's weird to turn this into a bullet list.

Reply:
removed the sub-list for this change